quil-py/src/instruction/gate.rs

```diff
@@ -149,46 +160,88 @@ impl PyPauliTerm {
     #[new]
     pub fn new(
         py: Python<'_>,
-        words: Vec<PyPauliGate>,
+        arguments: Vec<(PyPauliGate, String)>,
         expression: PyExpression,
-        arguments: Vec<Py<PyString>>,
     ) -> PyResult<Self> {
-        Ok(Self(
-            PauliTerm::new(
-                Vec::<PauliGate>::py_try_from(py, &words)?,
-                Expression::py_try_from(py, &expression)?,
-                Vec::<String>::py_try_from(py, &arguments)?,
-            )
-            .map_err(RustGateError::from)
-            .map_err(RustGateError::to_py_err)?,
-        ))
+        Ok(Self(PauliTerm::new(
+            Vec::<(PauliGate, String)>::py_try_from(
+                py,
+                &PyPauliTerm::py_pairs_from_tuples(py, arguments)?,
+            )?,
+            Expression::py_try_from(py, &expression)?,
+        )))
+    }
+
+    // Override the getters/setters generated by [`py_wrap_data_struct!`] so that they
+    // return/take tuples instead of the wrapping [`PyPauliPair`] type.
```

Comment:
Does this actually work? Huh.

Reply:
Yeah, a convenient, but not ideal, workaround. Added a test to make sure it works as we need it to. I also created [this issue](https://github.com/rigetti/rigetti-pyo3/issues/22) for `rigetti-pyo3`.